config/config-server/src/com/thoughtworks/go/service/ConfigRepository.java

```diff
@@ -390,8 +390,23 @@ void cleanAndResetToMaster() throws IOException {
             deleteBranch(BRANCH_AT_HEAD);
         } catch (Exception e) {
             String currentBranch = git.getRepository().getBranch();
-            LOGGER.error(format("Error while trying to clean up config repository, CurrentBranch: %s \n : \n Message: %s", currentBranch, e.getMessage(), e.getStackTrace()), e);
+            LOGGER.error("Error while trying to clean up config repository, CurrentBranch: {} \n : \n Message: {} \n StackTrace: {}", currentBranch, e.getMessage(), e.getStackTrace(), e);
             throw new RuntimeException(e);
         }
     }
+
+    public void garbageCollect() throws Exception {
+        doLocked(new VoidThrowingFn<Exception>() {
+            public void run() throws Exception {
+                try {
+                    LOGGER.info("Before GC: {}", git.gc().getStatistics());
+                    git.gc().setAggressive(systemEnvironment.get(SystemEnvironment.GO_CONFIG_REPO_GC_AGGRESSIVE)).call();
+                    LOGGER.info("After GC: {}", git.gc().getStatistics());
```

Comment:
`Before GC` and `After GC` could be confusing (heap GC?). Can we clarify this a bit?